Add message check to WebPageProxy::requestAttachmentIcon

pvollan opened the pull request:
#### 334993de44eed1ed8223454bc537a4124325e207
<pre>
Add message check to WebPageProxy::requestAttachmentIcon
<a href="https://bugs.webkit.org/show_bug.cgi?id=242694">https://bugs.webkit.org/show_bug.cgi?id=242694</a>
&lt;rdar://96832034&gt;

Reviewed by Chris Dumez.

WebPageProxy::requestAttachmentIcon should only be called if the attachment element is enabled. Also, passing a full pathname
when creating the icon for the attachment is not needed, it is sufficient to create a dummy filename with the same extension.

* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::requestAttachmentIcon):

Canonical link: <a href="https://commits.webkit.org/252533@main">https://commits.webkit.org/252533@main</a>
</pre>

cdumez commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -10443,8 +10445,15 @@ void WebPageProxy::requestAttachmentIcon(const String& identifier, const String&
     }
 #endif
 
+    String exampleFileName;
```
I added Wenson and Tim to confirm that it is indeed OK to only keep the extension part of the filename.

cdumez replied:
Should we initialize to `"example"_s` by default? This way it'd be "example" if we can't find an extension, which seems safer than passing a null String to iconForAttachment().

cdumez commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -10443,8 +10445,15 @@ void WebPageProxy::requestAttachmentIcon(const String& identifier, const String&
+    String exampleFileName;
+    size_t dotIndex = fileName.reverseFind('.');
+    if (dotIndex != notFound) {
+        String extension = fileName.substring(dotIndex);
```
We don't need to construct a temporary String here:
`auto extension = StringView(fileName).substring(dotIndex);`

cdumez commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -10443,8 +10445,15 @@ void WebPageProxy::requestAttachmentIcon(const String& identifier, const String&
+        exampleFileName = makeString("example", extension);
+    }
+
 #if PLATFORM(COCOA)
```
Shouldn't all the exampleFileName logic be inside this `#if` scope?

cdumez commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -10443,8 +10445,15 @@ void WebPageProxy::requestAttachmentIcon(const String& identifier, const String&
+
 #if PLATFORM(COCOA)
-    if (auto icon = iconForAttachment(fileName, contentType, title, size))
+    if (auto icon = iconForAttachment(exampleFileName, contentType, title, size))
```
I think this should be `exampleFilename`. AFAIK, `filename` is one word, not two.

pvollan replied:
> I think this should be `exampleFilename`. AFAIK, `filename` is one word, not two.

These are all very good points. I have updated the patch.

Thanks for reviewing!

cdumez commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -10444,7 +10446,15 @@ void WebPageProxy::requestAttachmentIcon(const String& identifier, const String&
+    StringBuilder exampleFilename;
+    exampleFilename.append("example"_s);
+    size_t dotIndex = fileName.reverseFind('.');
+    if (dotIndex != notFound) {
```
nit: could make a bit more concise IMO:
```
if (dotIndex != notFound)
    exampleFilename.append(StringView(fileName).substring(dotIndex));
```

cdumez reviewed (approved):
LGTM with nit.

pvollan committed:
Add message check to WebPageProxy::requestAttachmentIcon
https://bugs.webkit.org/show_bug.cgi?id=242694
<rdar://96832034>

Reviewed by Chris Dumez.

WebPageProxy::requestAttachmentIcon should only be called if the attachment element is enabled. Also, passing a full pathname
when creating the icon for the attachment is not needed, it is sufficient to create a dummy filename with the same extension.

* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::requestAttachmentIcon):

Canonical link: https://commits.webkit.org/252533@main
